hook/mutate_pod.go

```diff
@@ -113,23 +108,35 @@ func (m podMutator) Handle(ctx context.Context, req admission.Request) admission
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
 }
 
-func (m podMutator) targetStorageClasses(ctx context.Context) (map[string]storagev1.StorageClass, error) {
-	var scl storagev1.StorageClassList
-	if err := m.client.List(ctx, &scl); err != nil {
-		return nil, err
+type targetSC struct {
+	getter *getter.RetryMissingGetter
+	cache  map[string]*storagev1.StorageClass
+}
+
+func (t *targetSC) Get(ctx context.Context, name string) (*storagev1.StorageClass, error) {
+	if sc, ok := t.cache[name]; ok {
+		return sc, nil
 	}
 
-	targets := make(map[string]storagev1.StorageClass)
-	for _, sc := range scl.Items {
-		if sc.Provisioner != topolvm.PluginName {
-			continue
+	var sc storagev1.StorageClass
+	err := t.getter.Get(ctx, types.NamespacedName{Name: name}, &sc)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			t.cache[name] = nil
```

Comment:
This line is not executed when the cache contains target SC. I think it is very difficult to update the cache when an SC is deleted. I guess we have to accept read-after-create consistency miss for SC.

Reply:
This is not for consistency but for real no SC found case.
kubernetes accepts pod definitions regardless of the existence of depending SCs. In this case, we should not raise an error and continue the processing.

Reply:
By your reply, I recognized this block is not for cache consistency. And I misunderstood this step's meaning.